### PR TITLE
Remove deprecated features: `zenml deploy` and `zenml <stack-component> deploy`

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes
This PR removes the deprecated features that were used to deploy ZenML servers and ZenML stacks and stack components straight from the ZenML CLI. These are now replaced by one-click stack deployments and Terraform stack deployment modules. Mentions to the external mlstacks project are kept or even added as an alternative for now.

Docs are rendered here: https://zenml-io.gitbook.io/stefan-docs

### Breaking Changes

This PR removes the following CLI commands and the functionality associated with them:

* `zenml deploy`
* `zenml destroy`
* `zenml stack recipe *`
* `zenml stack deploy_mlstack`
* `zenml stack destroy`
* `zenml <stack-component> deploy`
* `zenml <stack-component> destroy`

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/how-to/trigger-pipelines/use-templates-dashboard.md [hunk 1]
```diff
@@ -27,9 +27,8 @@ In order to run a template from the dashboard:
 - You can go to a specific template page and click on `Run Template`.
 
 Either way, you will be forwarded to a page where you will see the 
-`Run Details`. Here, you have the option to upload a `.yaml` [configurations
-file](https://docs.zenml.io/how-to/use-configuration-files) or change the 
-configuration on the go by using our editor.
+`Run Details`. Here, you have the option to upload a `.yaml` [configurations file](../../how-to/use-configuration-files/README.md)
+or change the  configuration on the go by using our editor.
```
Comment (strickvl) [inline docs/book/how-to/trigger-pipelines/use-templates-dashboard.md:31]:
```suggestion
or change the configuration on the go by using our editor.
```

**Review by strickvl — CHANGES_REQUESTED** [review]
Good we're finally doing it. Some care will have to be take how we frame it in the release notes, but otherwise aside from the one nit I found, LGTM.

File: docs/book/getting-started/deploying-zenml/manage-the-deployed-services/custom-secret-stores.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+---
+description: Learning how to develop a custom secret store.
+---
+
+# Custom secret stores
+
```
Comment (wjayesh) [inline docs/book/getting-started/deploying-zenml/manage-the-deployed-services/custom-secret-stores.md:6]: can we move this to the same level as the secret-management.md file (that is, directly under deploying-zenml)? this section (`manage-the-deployed-services`) shouldn't exist anymore. in the `toc.md`, it is already positioned so.

**Review by wjayesh — CHANGES_REQUESTED** [review]
good to go save for a small restructuring

File: docs/book/getting-started/deploying-zenml/manage-the-deployed-services/custom-secret-stores.md (hunk 2 above)
Comment (wjayesh) [reply docs/book/getting-started/deploying-zenml/manage-the-deployed-services/custom-secret-stores.md:6]: you would then also have to update the path in the `toc.md` file to the parent directory. Thank you for fixing my oversight! :)

Comment (stefannica) [reply docs/book/getting-started/deploying-zenml/manage-the-deployed-services/custom-secret-stores.md]: Done :+1:

**Review by strickvl — APPROVED** [review]

**Review by htahir1 — APPROVED** [review]

**Review by wjayesh — APPROVED** [review]

**Pull request merged**